### DISCUSSION TITLE
@nuxt/kit module not working the same as expected in nuxt project

### DISCUSSION THREAD

**Discussion by gardun0** [nuxt/nuxt#16911] [Questions]
Hello,

I'm facing a couple of trouble trying to module our shared `components`, `plugins`, and `composables` for our projects.

## Problem 1

Within a project nuxt does import our composable at our server middlewares (using explicit import) but as a nuxt module, even if i try to explicit import the composables we use inside, it always throws

```
[nuxt] [request error] [unhandled] [500] useX is not defined
```

As far as i know, auto imports doesn't work on server, but im using explicit imports. The composable is used at the first line inside the `defineEventHandler` and I'm registering the handler like this:

```ts
const runtimeDir = fileURLToPath(new URL('./runtime', import.meta.url))

addServerHandler({
      handler: resolve(runtimeDir, 'middlewares/session'),
})
```

I think I've missed something cause, within a nuxt project, everything works fine.

## Problem 2

I have `layouts` in some nuxt project, everything works fine, but as soon as I register the layout within the module, stuff like `runtimeConfig` using `useRuntimeConfig` doesn't work, it seems `nuxtApp` can't initialize within a template registered by a module.

---

I hope i can have some feedback about what I'm missing, best regards ❤️ !

**Comment by danielroe** [comment]
It's difficult to say without seeing your code, but note:
* if you are injecting runtime files, you may need to add your runtime dir to `build.transpile`
* make sure you're not trying to use _vue composables_ in your server routes

**Reply by gardun0** [reply]
So, i think i'm using a composable inside a server route, but that works on server middleware within a nuxt project, otherwise how can i have an state on a server handler (acting as a global middleware).
